prometheus_client/exposition.py

```diff
@@ -185,13 +223,11 @@ def _use_gateway(method, gateway, job, registry, grouping_key, timeout):
     url = url + ''.join(['/{0}/{1}'.format(quote_plus(str(k)), quote_plus(str(v)))
                              for k, v in sorted(grouping_key.items())])
 
-    request = Request(url, data=data)
-    request.add_header('Content-Type', CONTENT_TYPE_LATEST)
-    request.get_method = lambda: method
-    resp = build_opener(HTTPHandler).open(request, timeout=timeout)
-    if resp.code >= 400:
-        raise IOError("error talking to pushgateway: {0} {1}".format(
-            resp.code, resp.msg))
+    headers=[('Content-Type', CONTENT_TYPE_LATEST)]
+    if handler is None:
```

Comment:
Not sure I can see how that would work, as the function is called with a handler of None by other functions in exposition.py. Do you mean make 'default_handler' the default instead of None for all of them?